plugins/modules/network/routeros/routeros_api.py

```diff
@@ -0,0 +1,481 @@
+        try:
+            arbitrary_result = self.api_path(arb_cmd, **param)
+            for i in arbitrary_result:
+                self.result['message'].append(i)
+            self.return_result(False)
+        except Exception as e:
+            self.errors(e)
+
+    def return_result(self, ch_status=False, status=True):
+        if not status:
+            self.module.fail_json(msg=to_native(self.result['message']))
+        else:
+            self.module.exit_json(changed=ch_status,
+                                  msg=self.result['message'])
+
+    def errors(self, e):
+        if e.__class__.__name__ == 'TrapError':
+            self.result['message'].append("%s" % e)
+            self.return_result(False, True)
+        self.result['message'].append("%s" % e)
+        self.return_result(False, False)
+
+
+def ros_api_connect(username, password, host, port, ssl, module):
+    # connect to routeros api
+    conn_status = {"connection": {"username": username,
+                                  "hostname": host,
+                                  "port": port,
+                                  "ssl": ssl,
+                                  "status": "Connected"}}
+    try:
+        if ssl is True:
+            if not port:
+                port = 8729
+                conn_status["connection"]["port"] = port
+            ctx = ssl.create_default_context()
+            ctx.check_hostname = False
+            ctx.set_ciphers('ADH:@SECLEVEL=0')
+            api = connect(username=username,
+                          password=password,
+                          host=host,
+                          ssl_wrapper=ctx.wrap_socket,
+                          port=port)
+        else:
+            if not port:
+                port = 8728
+                conn_status["connection"]["port"] = port
+            api = connect(username=username,
+                          password=password,
+                          host=host,
+                          port=port)
+    except Exception as e:
+        conn_status["connection"]["status"] = "error: %s" % e
+        module.fail_json(msg=to_native([conn_status]))
+
+    return api
+
+
+def main():
+    # define available arguments/parameters a user can pass to the module
+    module_args = (dict(
+        username=dict(type='str', required=True),
+        password=dict(type='str', required=True, no_log=True),
+        hostname=dict(type='str', required=True),
+        port=dict(type='int'),
+        ssl=dict(type='bool', default=False),
+        path=dict(type='str', required=True),
+        add=dict(type='str'),
+        remove=dict(type='str'),
+        update=dict(type='str'),
+        cmd=dict(type='str'),
+        query=dict(type='str')))
+
+    module = AnsibleModule(argument_spec=module_args,
+                           supports_check_mode=False)
+
+    if not HAS_LIB:
+        module.fail_json(msg=to_native('librouteros for Python is required for this module'))
```

Comment:
It's better to use https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/basic.py#L646 here instead of composing your own message. Also, you don't need `to_native()` for constant strings.

Reply:
fixed